core/blockchain.js

```diff
@@ -55,6 +56,152 @@ exports.handleTransaction = function handleTransaction(tx, block) {
     }
 };
 
+
+/**
+ * Get balance of account by address.
+ *
+ * @param   address            Address of account.
+ * @returns {Promise.<Number>} Balance of account.
+ */
+exports.getBalance = async function getBalance(address) {
+    const account = await getAccountFromLatestBlock(address);
+    return account.balance;
+};
+
+/**
+ * Get stake of account by address.
+ *
+ * @param   address            Address of account.
+ * @returns {Promise.<Number>} Stake of account.
+ */
+exports.getStake = async function getStake(address) {
+    const account = await getAccountFromLatestBlock(address);
+
+    if (!account.certificates.length) {
+        return 0;
+    }
+
+    return account.locked;
+};
+
+/**
+ * Get votes for account (delegate) by address.
+ *
+ * @param   address              Address of delegates.
+ * @returns {Promise.<String[]>} All votes for delegate.
+ */
+exports.getVotesFor = async function getVotesFor(address) {
+    const latestBlock = await chainData.getLatest();
+
+    const votes = latestBlock.state.filter(account => {
+        return account.votes.findIndex(delegate => delegate == address) >= 0;
+    });
+
+    return votes;
+};
+
+/**
+ * Get all delegates sorted by their votes.
+ *
+ * @returns {Promise.<Object[]>|Object[]} Delegates array.
+ */
+exports.getDelegates = async function getDelegates() {
+    let delegates = await getRawDelegates();
+
+    delegates = Object.keys(delegates).map(delegate => {
+        return {
+            delegate,
+            voteAmount: delegates[delegate]
+        };
+    });
+
+    delegates.sort((a,b) => b.voteAmount - a.voteAmount);
+
+    return delegates;
+};
+
+/**
+ * Returns 31 active delegates.
+ *
+ * @returns {Promise.<Object[]>}
+ */
+exports.getActiveDelegates = async function getActiveDelegates() {
+    const delegates = await this.getDelegates();
+
+    return delegates.slice(0, 31);
```

Comment:
31 is a magic number.
It has to be defined as module-level constant and exported outside.
i.e.
```JavaScript
const ACTIVE_DELEGATES_COUNT = exports.ACTIVE_DELEGATES_COUNT = 31;
```